Add QR Code Auth and various fixes

lillithrosepup opened the pull request:
I have added authentication from mobile QR code. The webpage will display the user profile picture and name before logging in, to verify it is the correct instance
<img width="960" alt="brave_VLaK61oCOX" src="https://user-images.githubusercontent.com/44179559/134994424-f5badf55-38ec-4edb-a572-01edb07cf1c9.png">
<img width="960" alt="brave_r3itUa3Nrn" src="https://user-images.githubusercontent.com/44179559/134994426-c8de58b5-3beb-4e70-8e06-19e8867bb02a.png">

lillithrosepup committed:
QR Code Auth + some fixes

ImLvna commented on src/App.vue:
```diff
@@ -74,7 +74,7 @@ export default {
         return "https://via.placeholder.com/256/2C2F33/FFFFFF?text=Loading";
       if (this.userData.avatar) {
-        return url + `avatars/${this.userData.id}/${this.userData.avatar}`;
+        return protocol + '://' + url + `avatars/${this.userData.id}/${this.userData.avatar}`;
```
I have also added some general enhancements to make it easier to code. This changes the 'url' config to the 'url' and 'protocol' variables, to make it easier to connect things such as websockets

lillithrosepup committed:
Fix error: Unexpected lexical declaration in case block (no-case-declarations)

lillithrosepup committed:
Join Server Icon

lillithrosepup committed:
Fix server sorting to discord's method, compare to actual discord client to verify.

lillithrosepup committed:
Fixed emoji and added partial image/vid support

lillithrosepup committed:
Forgot to remove a console.log

lillithrosepup committed:
Unused?

lillithrosepup committed:
Remove unused libraries and extra code

lillithrosepup committed:
Add optional token login

lillithrosepup committed:
Fix eslint errors

lillithrosepup committed:
Fix backend crashing with stage channels

lillithrosepup committed:
Accidentally gave a video a <img> tag

lillithrosepup committed:
Add 'Channel Memory'

On open, display the last opened channel.
If server is saved and no channel, default to first channel in server.
If no server is saved, defaut to opening latest DM

lillithrosepup committed:
Add Channel Categories

lillithrosepup committed:
clean up
